### DISCUSSION TITLE
Is it possible to add descriptions of transitive relationships for dependencies?

### DISCUSSION THREAD

**Discussion by zangcc** [aquasecurity/trivy#5025] [Ideas]
### Description

Hello trivy~ 🤝 I think trivy's scanning function has been done very well, and the scanning dependencies are very detailed, but there is a suggestion that I hope you can adopt.

First of all, this is the scan result of trivy, which can locate the file where the dependency is, for example:

![image](https://github.com/aquasecurity/trivy/assets/64825932/143f1d14-c34a-45fa-8100-fc8b1c1ad97e)
But it contains all the dependencies, many dependencies are transitive, we need to check the dependency graph for POM by ourselves.

So, my idea is can Trivy increase the declaration of the transitive relationship of dependencies?

Like the following example:

![image](https://github.com/aquasecurity/trivy/assets/64825932/e409b3e5-0c3a-47b2-a2fe-b5e9ab98e5d4)

Defect component commons-beanutils:commons-beanutils is introduced by org.apache.shiro:shiro-core component

Transitivity of dependencies can be appended to scan results. For example "Defective component commons-beanutils:commons-beanutils is introduced by org.apache.shiro:shiro-core component"

### Target

None

### Scanner

None

**Comment by ba1ma0** [comment]
Please add this feature！！！！！！ We need this ！ Thank you ～

**Comment by RexHarrr** [comment]
Hello, trivy~ I think it is necessary to add this function

**Comment by knqyf263** [comment]
We already have the feature. 
https://aquasecurity.github.io/trivy/v0.44/docs/configuration/reporting/#show-origins-of-vulnerable-dependencies

POM is not supported but is in progress.
https://github.com/aquasecurity/go-dep-parser/pull/243

**Reply by zangcc** [reply]
Ok 👏  I hope pom-maven projects can also support dependency trees as soon as possible. Will the next version support this feature? @knqyf263

**Reply by knqyf263** [reply]
It is not in the milestone, so we can't promise. But I believe it will be included in v0.46.0.
https://github.com/aquasecurity/trivy/milestone/23

**Reply by knqyf263** [reply]
Track https://github.com/aquasecurity/trivy/pull/4902
